lib/content-services/src/lib/category/services/category-tree-datasource.service.ts

```diff
@@ -56,25 +56,29 @@ export class CategoryTreeDatasourceService extends TreeService<CategoryNode>  {
             }
             const treeResponse: TreeResponse<CategoryNode> = {entries: nodesList, pagination: response.list.pagination};
             return treeResponse;
-        })) : this.categoryService.searchCategories(name, skipCount, maxItems).pipe(map((pagingResult) => {
+        })) : this.categoryService.searchCategories(name, skipCount, maxItems).pipe(mergeMap((pagingResult) => {
             const nextAfterGeneralPathPartIndex = 3;
             const pathSeparator = '/';
-            return {
-                entries: pagingResult.list.entries.map((category) => {
-                    const path = category.entry.path.name.split(pathSeparator).slice(nextAfterGeneralPathPartIndex)
-                        .join(pathSeparator);
-                    return {
-                        id: category.entry.id,
-                        nodeName: path ? `${path}/${category.entry.name}` : category.entry.name,
-                        parentId: category.entry.parentId,
-                        level: 0,
-                        nodeType: TreeNodeType.RegularNode,
-                        hasChildren: false,
-                        isLoading: false
-                    };
-                }),
-                pagination: pagingResult.list.pagination
-            };
-        }));
+            return from(pagingResult.list.entries).pipe(mergeMap((category) => {
+                const path = category.entry.path.name.split(pathSeparator).slice(nextAfterGeneralPathPartIndex)
+                    .join(pathSeparator);
+
+                return this.categoryService.getCategory(category.entry.id).pipe(
```

Comment:
This is giving me following error 
<img width="1723" alt="image" src="https://user-images.githubusercontent.com/115619721/233068885-4ee9037f-3e81-4386-8d3f-19aa45be7958.png">

Are not here missing changes for CategoryService?

Reply:
Yes, it will be added with this PR https://github.com/Alfresco/alfresco-ng2-components/pull/8470